From c2b4753970b7f7bf1bbee3dda54d5eba53c797d1 Mon Sep 17 00:00:00 2001
From: Werner Lemberg <wl@gnu.org>
Date: Tue, 3 Dec 2013 07:01:57 +0100
Subject: [PATCH 5/6] [truetype] Remove dead code.

Reported by Nigel Tao <nigeltao@golang.org>.

* include/internal/tttypes.h (TT_LoaderRec): Remove unused
`preserve_pps' field.
* src/truetype/ttgload.c (TT_Hint_Glyph): Updated.
---
 ChangeLog                  | 10 ++++++++++
 include/internal/tttypes.h |  1 -
 src/truetype/ttgload.c     | 11 ++++-------
 3 files changed, 14 insertions(+), 8 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 9ce39b6..59dc1f4 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,5 +1,15 @@
 2013-12-03  Werner Lemberg  <wl@gnu.org>
 
+	[truetype] Remove dead code.
+
+	Reported by Nigel Tao <nigeltao@golang.org>.
+
+	* include/internal/tttypes.h (TT_LoaderRec): Remove unused
+	`preserve_pps' field.
+	* src/truetype/ttgload.c (TT_Hint_Glyph): Updated.
+
+2013-12-03  Werner Lemberg  <wl@gnu.org>
+
 	[truetype] Fix phantom point handling.
 
 	This is a further improvement to the changes from 2013-11-06.
diff --git a/include/internal/tttypes.h b/include/internal/tttypes.h
index f4da3f9..ad302b8 100644
--- a/include/internal/tttypes.h
+++ b/include/internal/tttypes.h
@@ -1479,7 +1479,6 @@ FT_BEGIN_HEADER
     FT_Int           advance;
     FT_Int           linear;
     FT_Bool          linear_def;
-    FT_Bool          preserve_pps;
     FT_Vector        pp1;
     FT_Vector        pp2;
 
diff --git a/src/truetype/ttgload.c b/src/truetype/ttgload.c
index ca1f4b8..2c2bea5 100644
--- a/src/truetype/ttgload.c
+++ b/src/truetype/ttgload.c
@@ -832,13 +832,10 @@
 #endif
 
     /* save glyph phantom points */
-    if ( !loader->preserve_pps )
-    {
-      loader->pp1 = zone->cur[zone->n_points - 4];
-      loader->pp2 = zone->cur[zone->n_points - 3];
-      loader->pp3 = zone->cur[zone->n_points - 2];
-      loader->pp4 = zone->cur[zone->n_points - 1];
-    }
+    loader->pp1 = zone->cur[zone->n_points - 4];
+    loader->pp2 = zone->cur[zone->n_points - 3];
+    loader->pp3 = zone->cur[zone->n_points - 2];
+    loader->pp4 = zone->cur[zone->n_points - 1];
 
 #ifdef TT_CONFIG_OPTION_SUBPIXEL_HINTING
     if ( driver->interpreter_version == TT_INTERPRETER_VERSION_38 )
-- 
1.8.5.1

